### PR TITLE
fix(op-node): git commit print

### PR DESCRIPTION
**Pull request by shaspitz**
**Description**

When running `make devnet-up` on the develop branch, I encountered `panic: runtime error: slice bounds out of range [:8] with length 3`. I found the issue to be in a print statement within op-node's main.go. On the develop branch, `GitCommit` has a value of `"dev"`. 

**Tests**

Tested on my machine that `make devnet-up` now works.

**Additional context**

**Metadata**

### CODE REVIEW

File: op-node/cmd/main.go
Comment (ajsutton) [inline op-node/cmd/main.go]: I suspect we'd want to preserve the behaviour of omitting this if its empty, truncating if longer than 8 chars and otherwise including verbatim.

**Review by ajsutton — COMMENTED** [review]
We have this same pattern in `op-challenger/cmd/main.go` and `op-program/cmd/main.go` as well.

That said, I'm really surprised your getting `dev` as the value here. We run devnet tests in CI before merging any PR and they're passing (https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/30140/workflows/9fdd004c-b84b-4964-99cc-5a3ed6c5ccd6/jobs/1421207).  

The value for GitCommit is set from https://github.com/ethereum-optimism/optimism/blob/e02310c03927c5fbd2fcf5b30b516105b5497fb0/op-node/Makefile#L1. What do you get when you run `git rev-parse HEAD`?

**Comment by ajsutton** [comment]
I'm still surprised this is happening, but locally I do see this failure (once I got my local docker issues sorted...).  Very interesting that it passes in CI but fails locally. I'm guessing its related to https://github.com/ethereum-optimism/optimism/pull/7715

**Comment by ajsutton** [comment]
My inclination here btw is to fix the source of `GitCommit` since it should be the commit hash and not `dev`.  We may then also want to improve the parsing to avoid a panic but its useful to highlight the incorrect value we're getting currently.

**Comment by protolambda** [comment]
My bad, in the docker-compose I set the build-arg to "dev", since we don't want to load the whole `.git` dir (390MB on my local checkout that's only 2 weeks after a fresh install) into the docker build just to read a single commit hash. Specifying the commit-hash as build-argument is more ideal. But I didn't find a way to get docker-compose to set the argument dynamically from within the compose config.
In CI / makefile the build args, and thus the commit, will be set as expected.
I think we should update the devnet python setup to specify the ` --build-arg ` during the docker-compose build invocation.

**Review by protolambda — APPROVED** [review]
~merging this to unblock devnet issue. Will address git commit build arg in separate PR~ Added a commit to handle the issue @ajsutton raised, and added build-args to the docker compose invocation, so we don't need a follow-up PR anymore.

**Review by trianglesphere — APPROVED** [review]

File: bedrock-devnet/devnet/__init__.py
Comment (ajsutton) [inline bedrock-devnet/devnet/__init__.py]:
```suggestion
    git_date = subprocess.run(['git', 'show', '-s', "--format=%ct"], capture_output=True, text=True).stdout.strip()
```

There's no shell interpreting things so the single quotes were being included in the content resulting in:
```
t=2023-10-23T23:34:51+0000 lvl=info msg="Initializing rollup node"            version=v0.0.0-03c0061b-'1698096523'
```

File: op-program/host/cmd/main.go
Comment (tynes) [inline op-program/host/cmd/main.go:23]: This seems like it could be a standard function in `op-service` to dedupe code?

Comment (ajsutton) [reply op-program/host/cmd/main.go:23]: We could do that as a follow up, but I'm keen to get this in so the devnet works again.  Apparently an unresolved conversation drops the PR out of the merge queue again. :)

**Pull request merged**
